ProcessMaker/processmaker
# Implement Search Functionality for Public Screen Templates Tab

This PR introduces the search functionality for the Public Screen Templates tab. Users can now search for specific screen templates based on their names, descriptions, and category.

## Solution

- Update PublicTemplatesListing to include PMQL search functionality

## How to Test

1. Navigate to the Public Screen Templates tab.
2. Utilize the search input field to enter a query (e.g., screen template name or description).
3. Verify that the search results update dynamically as you type.
4. Try different search queries to ensure accurate and relevant results are displayed.

## Related Tickets & Packages
- [FOUR-13484](https://processmaker.atlassian.net/browse/FOUR-13484)

ci:next
. 

## Code Review Checklist


[FOUR-13483]: https://processmaker.atlassian.net/browse/FOUR-13483?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[FOUR-13484]: https://processmaker.atlassian.net/browse/FOUR-13484?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
